.github/workflows/contracts-deploy.yml

```diff
@@ -4,25 +4,55 @@ name: contracts-e2e-tests-and-deploy
 on:
   workflow_call:
 
-# DO NOT TOUCH THIS:
+# DO NOT TOUCH THIS !
+# cancelling workflow when contracts deployment is in progress will leave the environment in a bad state
 concurrency:
   group: ${{ github.workflow }}
   cancel-in-progress: false
 
 env:
-  CACHE_KEY: fe-benjamin-button
-  CONTRACTS_ENVFILE: fe-benjamin
-  NODE_VERSION: 16
-  S3BUCKET_PATH: contracts/fe-benjamin-button
+  CACHE_KEY: the-button
+  CONTRACTS_ENVFILE: devnet
+  NODE_VERSION: 18
+  CI_S3BUCKET_NAME: alephzero-devnet-eu-central-1-ci
+  S3BUCKET_PATH: contracts/devnet-button
 
 jobs:
-  build_and_deploy_contracts:
-    name: Deploy contracts on feature environment
+
+  changes:
+    name: Detect changes
+    runs-on: ubuntu-20.04
+    permissions:
+      pull-requests: read
+    outputs:
+      button: ${{ steps.filter.outputs.button }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2.3.4
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+           button:
+             - 'contracts/access_control/**'
+             - 'contracts/button/**'
+             - 'contracts/game_token/**'
+             - 'contracts/marketplace/**'
+             - 'contracts/simple_dex/**'
+             - 'contracts/ticket_token/**'
+             - 'contracts/wrapped_azero/**'
+             - 'workflows/contracts-deploy.yml'
+
+  build-and-deploy-contracts:
+    needs: [changes]
+    name: Deploy contracts on devnet
     if: >
-      (github.event_name == 'push' && github.ref_name == 'benjamin') ||
+      (needs.changes.outputs.button == 'true' &&
+      github.event_name == 'push' &&
+      github.ref_name == 'main') ||
       (github.event_name == 'pull_request' &&
```

Comment:
- first condition because this workflow has no control how it is called from other action 
- second condition so that you could use a label to force deployment from a pull request